mixins/provider-mixin.md

```diff
@@ -22,6 +22,9 @@ class InterestingFactProvider extends ProviderMixin(LitElement) {
 
 Once this has been set up, child components can request your provider's data via the `RequesterMixin` mixin, using the same key as the provider. Since the event that is generated to request a provider is bubbled until it is fulfilled, this means there can be an arbitrary number of components in the hierarchy between the provider and the requester, none of which have any knowledge of the data being requested or provided.
 
+NB: before the first render (i.e. in the constructor) the event handler won't be attached to the DOM, so in practice `this.requestInstance()` needs
```

Comment:
It sounds like the requestor could call `requestInstance` in the `connectedCallback` if it needs the dependency before first render.  Care ought to be taken however, because if the element is removed from the DOM and reconnected (ex. drag & drop), `connectedCallback` would be invoke again.

Reply:
That should be OK though... `requestInstance` doesn't actually add any event listeners, it just fires an event. So the only downside here would be that it would be called again and get a "fresh" value for whatever it was requesting.